addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -1,73 +1,219 @@
      */
-    init: function (el, data) {
-        this._super(el, data);
-        this.channelId = parseInt(data.channelId, 10);
-        this.canUpload = data.canUpload === 'True';
-        this.canPublish = data.canPublish === 'True';
-        console.log(this.channelId, this.canUpload, this.canPublish);
+    init: function (parent, options) {
+        options = _.defaults(options || {}, {
+            title: _t("Upload a document"),
+            size: 'medium',
+        });
+        this._super(parent, options);
+        this._setup();
+
+        this.channelID = parseInt(options.channelId, 10);
+        this.canUpload = options.canUpload === 'True';
+        this.canPublish = options.canPublish === 'True';
+
+        this.set('state', '_select');
+        this.on('change:state', this, this._onChangeType);
+        this.set('can_submit_form', false);
+        this.on('change:can_submit_form', this, this._onChangeCanSubmitForm);
+
         this.file = {};
-        this.index_content = '';
+        this.is_valid_url = true;
     },
-    /**
-     * @override
-     */
     start: function () {
-        this.$el.modal({
-            backdrop: 'static'
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self._resetModalButton();
         });
-        this._setCategoryId();
-        this._setTagIds();
-        return this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param {string} message
+     */
+    _alertDisplay: function (message) {
+        this._alertRemove();
+        $('<div class="alert alert-warning" role="alert">' + message + '</div>').insertBefore(this.$('form'));
+    },
+    _alertRemove: function () {
+        this.$('.alert-warning').remove();
+    },
+    /**
+     * Category and tags management from select2
+     *
+     * @private
+     */
+    _bindSelect2Dropdown: function () {
+        var self = this;
+        this.$('#category_id').select2(this._select2Wrapper(_t('Category'), false,
+            function () {
+                return self._rpc({
+                    route: '/slides/category/search_read',
+                    params: {
+                        fields: ['name'],
+                        domain: [['channel_id', '=', self.channelID]],
+                    }
+                });
+            })
+        );
+        this.$('#tag_ids').select2(this._select2Wrapper(_t('Tags'), true, function () {
+            return self._rpc({
+                route: '/slides/tag/search_read',
+                params: {
+                    fields: ['name'],
+                    domain: [],
+                }
+            });
+        }));
+    },
+    _fetchUrlPreview: function (url) {
+        return this._rpc({
+            route: '/slides/dialog_preview/',
```

Comment:
While we are at it, couldn't we rename that route ? dialog_preview is quite moche. Not important anyway, just passing by.

Reply:
yes it is moche. Does `slides/previwe_values/` is better? I have no really good idea ....